### PR TITLE
feat: remove as self in model 1.1

### PR DESCRIPTION
**Pull request by adriantam**
## Description
1. For model 1.1, do not use 'as self'.  Instead, always use ':'
2. Validate to ensure 1.0 still use 'as self' and contains no ':'

## References
Close https://github.com/openfga/syntax-transformer/issues/92

## Review Checklist
- [X] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [X] The correct base branch is being used, if not `main`
- [X] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by adriantam** [comment]
Incorrect column identified with
```
model
  schema 1.1
type user
type group
  relations
    define member:[user]
type resource
  relations
    define writer as self
    define reader as self or writer
```

**Comment by adriantam** [comment]
> Incorrect column identified with
> 
> ```
> model
>   schema 1.1
> type user
> type group
>   relations
>     define member:[user]
> type resource
>   relations
>     define writer as self
>     define reader as self or writer
> ```

Actually, confirmed that the proper line is highlighted

**Comment by adriantam** [comment]
> This error highlight looks incorrect, it should highlight `[user]`
> 
> image

Issue is now fixed
Screen Shot 2022-11-10 at 10 10 12 AM

File: src/openfga.ne [hunk 1]
```diff
@@ -39,23 +89,38 @@ define          ->  _newline define_initial (_spacing | _relation_types) _as _sp
                 type: 'single',
                 targets: [def]
             }
-        let allowedTypes = data[2] ? data[2][0] : [];
-        if (allowedTypes.length && typeof allowedTypes[0] !== "string") {
-          allowedTypes = [];
-        }
+        const allowedTypes = def.allowedTypes;
```
Comment (miparnisari) [inline src/openfga.ne:92]: question: won't this always be empty?

File: src/openfga.ne [hunk 2]
```diff
@@ -39,23 +89,38 @@ define          ->  _newline define_initial (_spacing | _relation_types) _as _sp
...
 	data => data[1]
 %}
 
-define_base  ->  (_naming | from_phrase) {%
+define_base  ->  ( _naming | from_phrase) {%
     data => {
 		const entry = data[0][0];
-		let target, rewrite, from
+		let target, rewrite, from;
+        const allowedTypes: string[] = [];
```
Comment (miparnisari) [inline src/openfga.ne:123]: Please fix indentation

File: tests/dsl.test.ts [hunk 3]
```diff
@@ -358,6 +401,44 @@ type T4
 type T5
   relations
     define A as A1 from A2 but not A3
+`);
+      const time2 = new Date();
+      expect(result.length).toEqual(1);
+      expect(time2.getTime() - time1.getTime()).toBeLessThan(1000);
```
Comment (miparnisari) [inline tests/dsl.test.ts:407]: i'm not 100% sure but i think `it('test name', (....), 1000)` achieves the same goal

**Comment by miparnisari** [comment]
I pasted the `openfga.ne` file in https://omrelli.ug/nearley-playground/ and i'm not able to get my simple model to be parsed correctly. There is an error in the console that may be cause but i can't find it

image

File: src/openfga.ne (hunk 1 above)
Comment (adriantam) [reply src/openfga.ne:92]: Yes, I will use empty array instead here.

File: src/openfga.ne (hunk 2 above)
Comment (adriantam) [reply src/openfga.ne:123]: done

File: tests/dsl.test.ts (hunk 3 above)
Comment (adriantam) [reply tests/dsl.test.ts:407]: done

**Review by miparnisari — APPROVED** [review]

**Pull request merged**